test/plugin-helpers.js

```diff
@@ -0,0 +1,267 @@
+import document from 'global/document';
+import videojs from 'video.js';
+import testDataManifests from './test-manifests.js';
+import {MediaSource} from 'videojs-contrib-media-sources';
+import sinon from 'sinon';
+
+// a SourceBuffer that tracks updates but otherwise is a noop
+export const MockSourceBuffer = videojs.extend(videojs.EventTarget, {
+  constructor() {
+    this.updates_ = [];
+
+    this.on('updateend', function() {
+      this.updating = false;
+    });
+    this.buffered = videojs.createTimeRanges();
+
+    this.duration_ = NaN;
+    Object.defineProperty(this, 'duration', {
+      get() {
+        return this.duration_;
+      },
+      set(duration) {
+        this.updates_.push({
+          duration
+        });
+        this.duration_ = duration;
+        this.updating = true;
+      }
+    });
+  },
+  appendBuffer(bytes) {
+    this.updates_.push({
+      append: bytes
+    });
+    this.updating = true;
+  },
+  remove(start, end) {
+    this.updates_.push({
+      remove: [start, end]
+    });
+    this.updating = true;
+  },
+
+  updating: false
+});
+
+export const MockMediaSource = videojs.extend(videojs.EventTarget, {
+  constructor() {},
+  duration: NaN,
+  seekable: videojs.createTimeRange(),
+  sourceBuffers: [],
+  addSeekableRange_(start, end) {
+    this.seekable = videojs.createTimeRange(start, end);
+  },
+  addSourceBuffer(mime) {
+    return new (videojs.extend(videojs.EventTarget, {
+      constructor() {},
+      abort() {},
+      buffered: videojs.createTimeRange(),
+      appendBuffer() {},
+      remove() {}
+    }))();
+  },
+  // endOfStream triggers an exception if flash isn't available
+  endOfStream(error) {
+    this.error_ = error;
+  },
+  open() {}
+});
+
+export const MockSourceBufferMediaSource = function() {
+  let mediaSource = new MediaSource();
+
+  mediaSource.addSourceBuffer = function(mime) {
+    let sourceBuffer = new MockSourceBuffer();
+
+    sourceBuffer.mimeType_ = mime;
+    this.sourceBuffers.push(sourceBuffer);
+    return sourceBuffer;
+  };
+
+  return mediaSource;
+};
+
+// patch over some methods of the provided tech so it can be tested
+// synchronously with sinon's fake timers
+let Helper = {
```

Comment:
This looks copy-pasted from the main test file (and it looks like we didn't get rid of it back there).